src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/unusedimports/InputUnusedImportsFromStaticMethodRef.java

```diff
@@ -0,0 +1,31 @@
+/*
```

Comment:
Please add inputs to this file:
1. Assignment of method reference, i.e. `Function f = Type::methodRef`, violation and ok
2. Constructor reference as method call argument and assignment, show this is ok
3. Method reference on generic type, violation and ok, in assignment and argument to method call
4. Violations on case where we have imported method in javadoc but only code usage is in method reference

Another input file, with similar inputs but with processJavadoc=false

Reply:
We need to be careful around `::` token, we can have type arguments as a child, like this:
```
  Supplier s1 = C::<Integer>m;
  Supplier s2 = C::<@I Integer>m;

```
Also, please add method reference with fully qualified name, and show that we still place a violation on unused import, example:
```
 test.map(java.util.Objects::nonNull);
```

We should still place a violation on this import:
```
import java.util.Objects;

```

Reply:
If we are running out of methods/classes, we can create another test/ test input if we need to.

Reply:
Ah, Its getting really tricky. Thanks for your feedback. I will check these.

Reply:
@renjith7189 please ping m when you have added all cases and are ready to continue

Reply:
@nrmancuso I have added all the tests I could think of. Could you please check again?

Reply:
Ok, one last consideration for test inputs: it looks like we extract javadoc inline tags and regular tags and add them to references. Please create a new input file and test with method references only in javadocs (not in code), and show that we completely ignore the following invalid javadoc tags:
```
    /**
     * Use {@link Collections::emptyEnumeration} in this javadoc.
     * Use {@link Collections::emptyEnumeration} in this javadoc.
     * @see Collections::emptyEnumeration
     * @throws IllegalAccessError::new
     */
    public static void m() {}
```